Create wrapper functions

emmakatherina opened the pull request:
Fixes issue #70 

Please review and/or merge.

emmakatherina commented on gap/main.gi:
```diff
@@ -0,0 +1,64 @@
+        result := DoEchelonMatTransformationBlockwise( mat );
+        return rec(
+            vectors := result.vectors,
+            heads := result.pivotrows, #TODO ersetze mit heads
```
This is in the other PR.

ssiccha replied:
The other PR was merged so please update this. Or am I missing something?

ssiccha replied:
```suggestion
            heads := result.heads,
```

emmakatherina commented:
I am not sure whether the commit history is correct (the commit 690ea6c is originally from another PR but after the new commits). But maybe this is just a different order and we don't get commits twice, now?

ssiccha commented:
I guess the extra commit will disappear once we rebase this on top of master. Weird that it only got added "after" the README change.

emmakatherina commented:
Is there anything speaking against merging the code now? Did I miss something?

ssiccha commented:
No, you didn't miss something. I just didn't have the time yet to look through the whole PR. Sorry.

ssiccha commented:
IMO the commit title `Create global functions` should be changed to `Create wrapper functions`. `global function` is a very general description and could mean anything. You can change this by doing `git rebase -i master` and choose the reword option.

Another minor point: you amended the requested changes into the last commit. It would have been nicer if you put them into the first commit. You can merge a commit into an arbitrary other commit by doing `git rebase -i master`. What I usually do is:
- create commits with the requested changes and call them
  - `SQUASH: ...`
  - `SQUASH: ...`
  - ...
- Then call `git rebase -i master` which opens an editor. There you move each new commit under the respective commit that you want to merge it into. If you would quit now this would reorder the commits. Don't quit yet though. Select the `squash` instead of the `pick` option for the new commits. This lets you merge them into the commit directly above them.

ssiccha commented on gap/main.gd:
```diff
@@ -0,0 +1,5 @@
+# Declare global functions of this package
+
+DeclareGlobalFunction( "DoEchelonMatTransformationBlockwise", "Calculates echelon form and all other informations" );
```
```suggestion
DeclareGlobalFunction( "DoEchelonMatTransformationBlockwise", "Calculates echelon form and all other information" );
```
"Information" has no plural form in English.

ssiccha commented on gap/main.gi:
```diff
@@ -0,0 +1,67 @@
+            numberChopsHeight := options[3];
+            numberChopsWidth := options[4];
+        fi;
+        Info(InfoGauss, 1, "The matrix has a height of ", dim[1],
```
IMO we should only print the number of chops in both directions. The user already knows the size of the matrix. Unnecessary extra information tends to make the important information less visible.

Also please add an extra info statement that warns the user if any of `numberChops..` is equal to 1. In that case I think we will get almost no speed gain by using multiple threads.

emmakatherina commented:
Thank you very much for your detailed explanation of the rebase command! I just learned a lot of usefull stuff. Unfortunately git does complain about merge conflicts, so I added the new changes to the last commit anyway...

The commit message is also changed.

ssiccha commented:
Yeah, I wouldn't try to resolve all these merge conflicts then. That way lies madness! :)
